proto/src/serializers/optional.rs

```diff
@@ -1,6 +1,9 @@
 //! Serialize/deserialize `Option<T>` type where `T` has a serializer/deserializer.
-//! Use `null` if the received value equals the `Default` implementation.
-// Todo: Rename this serializer to something like "default_eq_none" to mirror its purpose better.
+//! Deserialize to `None` if the received value equals the `Default` value.
+//! Serialize `None` as `Some` with the `Default` value for `T`.
```

Comment:
The behaviour of this helper is questionable, before or after this change. If the type has a reasonable default value, why make the domain type field an `Option`? And in the other direction, why uselessly serialize a default value when a `None` would do as well?

Reply:
Yes, It's kind of a head-scratcher!
I see this is derived on fields of certain proto types with `Option<…>` types (e.g. [here](https://github.com/informalsystems/tendermint-rs/blob/bdef9884221a850323f98b3fa2b9b7471f5e8437/proto/src/prost/v0_34/tendermint.types.rs#L200-L201)). Perhaps, those types do not inherently need to be Option. Consequently, they required serde to handle Option this way.